tests/src/Model/MetadataTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenEuropa\Tests\EuropaSearchClient\Model;
+
+use OpenEuropa\EuropaSearchClient\Model\Metadata;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Tests the Metadata model class.
+ *
+ * @covers \OpenEuropa\EuropaSearchClient\Model\Metadata
+ */
+class MetadataTest extends TestCase
+{
+    /**
+     * Tests the setters and getters.
+     */
+    public function testSettersAndGetters(): void
+    {
+        $model = new Metadata();
+        $model->setCollection([
+            'BODY_VALUE' => ["Value 1", "Value 2"],
+        ]);
+        $this->assertEquals($model->getCollection(), ['BODY_VALUE' => ["Value 1", "Value 2"]]);
```

Comment:
Let's use `::assertSame()`. The difference is that `::assertSame()` is strict comparison (===), while `::assertEquals()` is non-strict (==). Also, in PHPUnit, the expectation comes first.